config/configgrpc/configgrpc.go

```diff
@@ -152,6 +152,9 @@ type GRPCServerSettings struct {
 
 	// Auth for this receiver
 	Auth *configauth.Authentication `mapstructure:"auth,omitempty"`
+
+	// Include propagates the incoming connection's metadata to downstream consumers.
+	IncludeMetadata bool `mapstructure:"include_metadata,omitempty"`
```

Comment:
Do we want users to be able to select which "metadata" to copy from the request? Maybe not now, but we can be safe and mark this experimental for the moment.

Reply:
Certainly doable, blindly copying all of it could be bad for memory if the request is decorated with large metadata? But that would also mean we'd have to roll our own `md.Copy()` or `header.Clone()` routines, as copying-first-then-filtering would be pointless in terms of memory.

Reply:
That's why I recommend just adding a comment that this is experimental, and we should look a bit into how/if we want this.

Reply:
@bogdandrutu added